metaseq/cli/train.py

```diff
@@ -112,24 +112,32 @@ def main(cfg: DictConfig) -> None:
     # Print args
     logger.info(cfg)
 
-    # Setup task, e.g., translation, language modeling, etc.
-    task = tasks.setup_task(cfg.task)
-
     assert cfg.criterion, "Please specify criterion to train a model"
 
     # Build model and criterion
     if cfg.distributed_training.ddp_backend == "fully_sharded":
         extra = {
             "use_sharded_state": cfg.distributed_training.use_sharded_state,
         }
+        memory_efficient_fp16 = cfg.distributed_training.memory_efficient_fp16
+        fp32_reduce_scatter = cfg.distributed_training.fp32_reduce_scatter
+        cfg.distributed_training.memory_efficient_fp16 = cfg.distributed_training.fp16
+        cfg.distributed_training.fp32_reduce_scatter = not cfg.distributed_training.fp16
```

Comment:
These should be flags that are set in the sweeps, instead of clobbered / derived here :(

Reply:
oh wow I totally missed the point of this! ok - let me.. just add some comments to explain this since i'll probably also forget this reason ^ if i ever look at this bit of code again 😅

Reply:
done - added a comment. also, I added a flag `cfg.distributed_training.task_ddp_backend` which be default wont wrap the task at all (keeping things as is).